rootchain/sampool/bucket.go

```diff
@@ -0,0 +1,63 @@
+package sampool
+
+import (
+	"github.com/0xPolygon/polygon-edge/rootchain"
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+type samSet struct {
+	messages   []rootchain.SAM
+	signatures map[string]bool
```

Comment:
I think it's just used to hold flags telling which messages are stored. You might want to rename it to some proper name like `isAdded`?

Reply:
[Renamed](https://github.com/0xPolygon/polygon-edge/pull/718/commits/6af174b0e8143c30e9b4f92e54fccbfe61030dcf)